lib/archive.js

```diff
@@ -142,7 +142,7 @@ Archive.prototype = {
         return void callback.call( this, error )
       }
 
-      if( ( folder.compressionType & 0xFF00 ) !== Cabinet.COMPRESSION.NONE ) {
+      if( ( folder.compressionType & 0x0F ) !== Cabinet.COMPRESSION.NONE ) {
```

Comment:
Compression algorithm is the full lower 8 bits, so the mask here should be `0xFF` then, I think
```suggestion
      if( ( folder.compressionType & 0xFF ) !== Cabinet.COMPRESSION.NONE ) {
```